File: 2022/04/4.py
silver = gold = 0
for line in open("input.txt").readlines():
    a, b, c, d = map(int, line.replace("-", ",").split(","))
    cover = lambda w, x, y, z: w <= y and x >= z
    silver += cover(a, b, c, d) or cover(c, d, a, b)
    gold += max(a, c) <= min(b, d)

print(silver, gold)

File: 2022/04/4.rb
silver = gold = 0
File.readlines("input.txt", chomp: true).each{
  a, b = eval "[#{_1.gsub /\-/, '..'}]"
  silver += 1 if a.cover? b or b.cover? a
  gold   += 1 if (a.to_a & b.to_a).any?
}

puts [silver, gold]*", "
